plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -898,6 +978,8 @@ void producer_plugin::set_program_options(
           "time in microseconds the write window lasts")
          ("read-only-read-window-time-us", bpo::value<uint32_t>()->default_value(my->_ro_read_window_time_us.count()),
           "time in microseconds the read window lasts")
+         ("allow-ro-trx-on-producer-node", bpo::value<bool>()->default_value(false),
```

Comment:
We should not have this as an option. This is for testing only.

Reply:
Thanks Lin. I don't see another way to disable the checking for producer, so we can run the read window in the C++ test `test_read_only_trx.cpp`. If you can suggest a better solution, I'll be glad to implement it.

Reply:
You could create a header method `verify_no_producers(my->_producers.empty(), ""--read-only-threads not allowed on producer node"")` that had a `#ifdef`. You could add a static variable that you set before construction and check in `plugin_initialize`.

Reply:
I asked Areg on telegram. If he prefers not to add an option I'll go with the static variable.

Reply:
I switched to using a static variable as suggested by Kevin.